plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/nessie/TestIcebergNessieCatalogWithAuth.java

```diff
@@ -0,0 +1,117 @@
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestIcebergNessieCatalogWithAuth
+        extends AbstractTestQueryFramework
+{
+    private Path tempDir;
+
+    private String nessieUri;
+
+    @AfterClass(alwaysRun = true)
+    public void teardown()
+            throws IOException
+    {
+        deleteRecursively(tempDir, ALLOW_INSECURE);
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Network network = Network.newNetwork();
+
+        KeycloakContainer keycloakContainer =
+                closeAfterClass(KeycloakContainer.builder().withNetwork(network).build());
+        keycloakContainer.start();
+
+        ImmutableMap<String, String> envVars = ImmutableMap.<String, String>builder()
+                .putAll(NessieContainer.defaultEnvVars())
+                .put("QUARKUS_OIDC_AUTH_SERVER_URL", KeycloakContainer.getUrlWithHostName() + "/realms/master")
+                .put("QUARKUS_OIDC_CLIENT_ID", "projectnessie")
+                .put("NESSIE_SERVER_AUTHENTICATION_ENABLED", "true")
+                .buildOrThrow();
+
+        NessieContainer nessieContainer =
+                closeAfterClass(NessieContainer.builder().withEnvVars(envVars).withNetwork(network).build());
+        nessieContainer.start();
+        nessieUri = nessieContainer.getRestApiUri();
+
+        tempDir = Files.createTempDirectory("test_trino_nessie_catalog");
+
+        return getDistributedQueryRunner(nessieUri, true, keycloakContainer.getAccessToken());
+    }
+
+    private DistributedQueryRunner getDistributedQueryRunner(String nessieUri, boolean isBearerAuth, String accessToken)
+            throws Exception
+    {
+        ImmutableMap.Builder<String, String> properties = ImmutableMap.builder();
+        properties.put("iceberg.file-format", "PARQUET")
+                .put("iceberg.catalog.type", "nessie")
+                .put("iceberg.nessie-catalog.uri", nessieUri)
+                .put("iceberg.nessie-catalog.default-warehouse-dir", tempDir.toString())
+                .put("iceberg.writer-sort-buffer-size", "1MB");
+        if (isBearerAuth) {
+            properties.put("iceberg.nessie-catalog.authentication.type", "BEARER");
+        }
+        if (accessToken != null) {
+            properties.put("iceberg.nessie-catalog.authentication.token", accessToken);
+        }
+
+        return IcebergQueryRunner.builder()
+                .setBaseDataDir(Optional.of(tempDir))
+                .setIcebergProperties(properties.buildOrThrow())
+                .build();
+    }
+
+    @Test
+    public void testWithInvalidAccessToken()
+    {
+        assertThatThrownBy(() -> getDistributedQueryRunner(nessieUri, true, "invalid_token"))
+                .isInstanceOf(QueryFailedException.class)
+                .hasMessageContaining("Unauthorized (HTTP/401)");
```

Comment:
>  Can we skip schema initialization and verify the query failure explicitly?

I am not sure how to skip schema initialisation. Can you please point me to some existing example/code?

> The failed place is unclear from this assertion. I know it comes from CREATE SCHEMA

Yes. It is while creating a namespace from Nessie catalog during create schema. 
```
at org.apache.iceberg.nessie.NessieIcebergClient.createNamespace(NessieIcebergClient.java:181)
	at io.trino.plugin.iceberg.catalog.nessie.TrinoNessieCatalog.createNamespace(TrinoNessieCatalog.java:135)
	at io.trino.plugin.iceberg.IcebergMetadata.createSchema(IcebergMetadata.java:804)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.createSchema(ClassLoaderSafeConnectorMetadata.java:398)
	at io.trino.tracing.TracingConnectorMetadata.createSchema(TracingConnectorMetadata.java:337)
	at io.trino.metadata.MetadataManager.createSchema(MetadataManager.java:720)```

Reply:
> I am not sure how to skip schema initialisation. Can you please point me to some existing example/code?

Please take a look at `schemaInitializer` in IcebergQueryRunner.

Reply:
> Can we skip schema initialization

I think we cannot skip schema initialization. default `SchemaInitializer` will create `tpch` schema atleast and I cannot set null from `IcebergQueryRunner.builder().setSchemaInitializer(null)`.

Also note that Namespaces (schema) is mandatory for Nessie table creation. So, I cannot manually create them before hand as it will fail too.

Reply:
Also I don't think it is a deal breaker as the final callstack is from Nessie and test is failing as expected.

Reply:
`HiveQueryRunner` has an option called `setCreateTpchSchemas` to disable schema initialization. 
But that is not there for `IcebergQueryRunner` . Thats why I got confused.

Do you want me to enhance `IcebergQueryRunner` to support disabling the schema initialization? I think it can be done in follow up. I don't want divert from the scope of this PR.

Reply:
Let's just remove `testWithInvalidAccessToken`. It's not so helpful.

Reply:
ok. Removed.